exchanges/graphcache/src/store/data.ts

```diff
@@ -554,25 +538,53 @@ export const inspectFields = (entityKey: string): FieldInfo[] => {
   return fieldInfos;
 };
 
+export const persistData = () => {
+  if (currentData!.storage) {
+    const entries: SerializedEntries = makeDict();
+    currentIgnoreOptimistic = true;
+    currentData!.persist.forEach(key => {
+      const sepIndex = indexOfSeparator(key);
+      if (sepIndex > -1) {
+        const entityKey = key.slice(0, sepIndex);
+        const fieldKey = key.slice(sepIndex + 1);
+        let x: void | Link | EntityField;
+        if ((x = readLink(entityKey, fieldKey)) !== undefined) {
+          entries[key] = `:${stringifyVariables(x)}`;
+        } else if ((x = readRecord(entityKey, fieldKey)) !== undefined) {
+          entries[key] = stringifyVariables(x);
+        } else {
+          entries[key] = undefined;
+        }
+      }
+    });
+
+    currentIgnoreOptimistic = false;
+    currentData!.storage.write(entries);
+    currentData!.persist.clear();
+  }
+};
+
 export const hydrateData = (
   data: InMemoryData,
   storage: StorageAdapter,
   entries: SerializedEntries
 ) => {
   initDataState(data, null);
+
   for (const key in entries) {
-    const dotIndex = key.indexOf('.');
-    const entityKey = key.slice(2, dotIndex);
-    const fieldKey = key.slice(dotIndex + 1);
-    switch (key.charCodeAt(0)) {
-      case 108:
-        writeLink(entityKey, fieldKey, entries[key] as Link);
-        break;
-      case 114:
-        writeRecord(entityKey, fieldKey, entries[key]);
-        break;
+    const value = entries[key];
+    const sepIndex = indexOfSeparator(key);
+    if (value && sepIndex > -1) {
+      const entityKey = key.slice(0, sepIndex);
+      const fieldKey = key.slice(sepIndex + 1);
+      if (value[0] === ':') {
```

Comment:
Nit: Could ternary this to save a couple of bytes and avoid nested if

Reply:
Closure & Terser should both ternary this automatically :)

Reply:
Good to know! 🚀